grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/query/ReasonerQueryImpl.java

```diff
@@ -178,7 +175,7 @@ private boolean isTransitive() {
     }
 
     boolean isAtomic() {
-        return selectAtoms().size() == 1;
+        return getAtoms().stream().filter(Atomic::isSelectable).count() == 1;
```

Comment:
Question: Does `getAtoms()` get very large? I ask because I often see `isAtomic()` and if calling this means iterating over a large set over and over again then performance may get affected.